src/main/java/seedu/letsgethired/logic/parser/NoteCommandParser.java

```diff
@@ -20,22 +23,31 @@ public class NoteCommandParser implements Parser<NoteCommand> {
      */
     public NoteCommand parse(String args) throws ParseException {
         requireNonNull(args);
-        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NOTE);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NOTE_INSERT, PREFIX_NOTE_DELETE);
 
         Index index;
 
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            index = ParserUtil.parseApplicationIndex(argMultimap.getPreamble());
         } catch (ParseException pe) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, NoteCommand.MESSAGE_USAGE), pe);
         }
 
-        if (!argMultimap.getValue(PREFIX_NOTE).isPresent()) {
-            throw new ParseException(NoteCommand.NO_NOTE_PARAMETER_MESSAGE);
+        String maybeInsert = argMultimap.getValue(PREFIX_NOTE_INSERT).orElse(null);
+        String maybeDelete = argMultimap.getValue(PREFIX_NOTE_DELETE).orElse(null);
+
```

Comment:
Minor nit: perhaps we can follow the other commands and use `.isPresent()` instead?